woocommerce/google-listings-and-ads
# Opt-in all `@wordpress` packages imports via DEWP

### Changes proposed in this Pull Request:

It's the final part of #1833

To opt-in all `@wordpress` packages imports via DEWP and revert all gradually externalized `@wordpress/components` imports, this PR:

- Remove the `extracted/` prefix from all `'extracted/@wordpress/components'` imports.
- Opt-in `@wordpress/components` and `@wordpress/compose` to import from the external source of DEWP.
   - Remove them from the `requestToExternal` array in the webpack config. So that all their imports will be externalized via regular DEWP instead of the mechanism of `extracted/` prefix.
   - The `@wordpress/compose` package was removed together as the following uses don't have external changes after `@wordpress/compose` [5.1.2 (used by WC 6.9)](https://github.com/woocommerce/woocommerce/blob/6.9.0/packages/js/components/package.json#L39): 
      - `createHigherOrderComponent`
      - `useFocusOutside`: refactor to TypeScript
      - `useInstanceId`: refactor to TypeScript
      - `useReducedMotion`
      - `useViewportMatch`
      - [Changelog](https://github.com/WordPress/gutenberg/blob/wp/6.3/packages/compose/CHANGELOG.md)
- Remove a compatible workaround for `DatePicker` as it has been fixed since WC 7.2.
   - Considering the latest WC version is way beyond the current L-2 version, it should be fine to remove this WC 7.0-7.1 workaround.
   - The original issue: #1739
   - Fixed by https://github.com/woocommerce/woocommerce/pull/35649
- Rebalance the max size config of bundlewatch for each non-wildcard file to 1.2 times.
- Update the .externalized.json file.
- Revert "Prevent the "import/no-duplicates" warning for the externalization process". (see [#1844](https://github.com/woocommerce/google-listings-and-ads/pull/1844))

### Screenshots:

![image](https://github.com/woocommerce/google-listings-and-ads/assets/17420811/343a8d9d-82dc-4234-ac74-4043ac229815)

### Detailed test instructions:

1. Check if all build scripts, `npm start`, `npm run dev`, `npm run start:hot` and `npm run build`, can run successfully.
2. Run bundle analyzer: `npm run env -- NODE_ENV=production wp-scripts build --webpack-bundle-analyzer`
   - Check if there is no DEWPable `@wordpress/*` package included in the build bundle.
   - P.S. `@wordpress/icons` is not a DEWPed package. [Ref](https://github.com/WordPress/gutenberg/blob/%40wordpress/dependency-extraction-webpack-plugin%404.4.0/packages/dependency-extraction-webpack-plugin/lib/util.js#L2)
3. Search the codebase to check if there is no `'extracted/@wordpress/components'` import.

### Changelog entry
